crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -2515,3 +2517,51 @@ fn range_size_hint(range: &impl RangeBounds<TxNumber>) -> Option<usize> {
     };
     end.checked_sub(start).map(|x| x as _)
 }
+
+/// Limits on how long pruning can run before it's forced to stop and thereby yield the
+/// [`DatabaseProviderRW`] hook.
+#[derive(Debug, Default, Clone, Copy, Constructor)]
+pub struct PruneLimit {
+    /// Maximum entries to delete from the database in one prune job.
+    segment_limit: Option<usize>,
```

Comment:
this shouldn't be a limit per segment, it's a limit for the number of entries that were deleted in one Pruner run